extensions/setname.md

```diff
@@ -0,0 +1,112 @@
+---
+title: IRCv3 `setname` Extension
+layout: spec
+copyrights:
+  -
+    name: "Janne Mareike Koschinski"
+    period: "2019"
+    email: "janne@kuschku.de"
+---
+
+## Notes for implementing work-in-progress version
+
+This is a work-in-progress specification.
+
+Software implementing this work-in-progress specification MUST NOT use the
+unprefixed `setname` capability name. Instead, implementations SHOULD use
+the `draft/setname` capability name to be interoperable with other software
+implementing a compatible work-in-progress version.
+
+The final version of the specification will use an unprefixed capability name.
+
+## Motivation
+
+Realnames have historically been immutable. This has never been much of an
+issue, as they were rarely used and usually not as present in client UIs.
+Still, multiple IRCds have provided independent, non-standard
+commands to update realnames.
+
+Nowadays, this has changed. Multiple clients show realnames directly in chat
+(e.g. next to the nickname) or use information from the realname to provide
+user-visible enhancements (e.g. using emails in the realname to provide
+gravatar based avatars).
+
+Due to this, the ability to change the realname has become relevant.
+
+## Description
+
+The `draft/setname` client capability allows clients to change their realname
+(GECOS). It also allows servers to directly inform clients about such a change.
+This avoids clients having to actually disconnect and reconnect. 
+
+This capability MUST be referred to as `draft/setname` at capability
+negotiation time.
+
+When enabled, clients are allowed to send a `SETNAME` command to notify servers
```

Comment:
If the cap isn't enabled, can the client not send a `SETNAME` command? If the client can't use the command, should the command `FAIL` with a more general error code like `CAP_NOT_NEGOTIATED` - message of `"Capability 'draft/setname' is required to use this command, and is not negotiated"` maybe?

Reply:
No, client can always send the command, but won't get a response.

Reply:
Whichever way is intended, spec language should be clarified to spell it out

Reply:
Considering the way IRCds implement it currently, I’d suggest that clients can always send it, but they won’t get a response (as unreal/inspircd already implement it).

Reply:
Slight nibble: The sentence here says _"When enabled, clients are allowed to send a `SETNAME` command..."_. If it's actually able to send the command all the time, should we change this sentence to reflect that? I like the clarification below, makes sense.